### DISCUSSION TITLE
Components inside Components in view! {}

### DISCUSSION THREAD

**Discussion by BigfootN** [Relm4/Relm4#679] [Q&A]
Hello,

Imagine I have a component `Foo` and a component `Bar`.

Is it possible to place / add / reference `Foo` inside the component `Bar` ? Is it possible to place it in the `view!` macro ?

If yes how to ? And why isn't it mentioned in the [book](https://relm4.org/book/stable/introduction.html) (from what I've read) ?

Or are the `template` the only way to do this ?

Thank you very much in advance for any help

**Comment by sanpii** [comment]
Hello,

The way I found (I don’t understand how to use widget templates…):

```rust
#[relm4::component]
impl SimpleComponent for Bar {
    type Init = ();
    type Input = ();
    type Output = ();

    view! {
        gtk::Box {
            append: model.foo.widget(),
        }
    }

    fn init(
        counter: Self::Init,
        root: Self::Root,
        sender: ComponentSender<Self>,
    ) -> ComponentParts<Self> {
        let foo = Foo::builder()
            .launch(())
            .detach();

        let model = AppModel { foo };

        let widgets = view_output!();

        ComponentParts { model, widgets }
    }
}
 ```

It would be nice to have a syntax similar to relm3 to do this directly in the `view!` macro:

```rust
view! {
    gtk::Box {
        Foo(/* init */) {
            /* Messages forwarding */
        }
    }
}
```
